src/com/liferay/portal/ejb/UserManagerImpl.java

```diff
@@ -430,9 +429,15 @@ public void sendPassword(String companyId, String emailAddress, Locale locale)
 
 		Company company = CompanyUtil.findByPrimaryKey(companyId);
 
-		String url=(company.getPortalURL().contains("://") ? "" : "https://") +
-		        company.getPortalURL() + "/c/portal_public/login?my_account_cmd=ereset&my_user_id="+user.getUserId()+
-		        "&token="+token+"&switchLocale="+locale.getLanguage()+"_"+locale.getCountry();
+		String url = null;
+		String urlPrefix = (company.getPortalURL().contains("://") ? "" : "https://") + company.getPortalURL();
+
+		if ( !fromAngular ) {
+			url = urlPrefix + "/c/portal_public/login?my_account_cmd=ereset&my_user_id=" + user.getUserId() +
+					"&token=" + token + "&switchLocale=" + locale.getLanguage() + "_" + locale.getCountry();
+		}else{
+			url = urlPrefix + "/" + java.text.MessageFormat.format("html/ng?resetPassword=true&userId={0}&token={1}", user.getUserId(), token);

```

Comment:
I would prefer to use an strategy to build the Url such as UrlStrategy instead of just passing a parameter, could be more OOP